app/router/student.py

```diff
@@ -1,384 +1,309 @@
-            status_code=400, detail="Student ID could not be generated. Max loops hit!"
-        )
-
-
-@router.get("/")
-def get_students(request: Request):
-    """
-    This API returns a student or a list of students who match the criteria(s) given in the request.
+    raise HTTPException(
+        status_code=400, detail="JNV Student ID could not be generated. Max loops hit!"
+    )
 
-    Optional Parameters:
-    student_id (str), birth_date (str), category (str), stream (str).
-    For extensive list of optional parameters, refer to the DB schema note on Notion.
 
-    Returns:
-    list: student data if student(s) whose details match, otherwise 404
+async def create_school_user_record(data, school_name):
+    school_data = school.get_school(build_request(query_params={"name": school_name}))
+    group_data = group.get_group(
+        build_request(query_params={"child_id": school_data["id"], "type": "school"})
+    )
 
-    Example:
-    > $BASE_URL/student/
-    returns [data_of_all_students]
+    await group_user.create_group_user(
+        build_request(
+            method="POST",
+            body={"group_id": group_data[0]["id"], "user_id": data["user"]["id"]},
+        )
+    )
+    enrollment_record_data = {
+        "academic_year": str(datetime.now().year)
+        + "-"
+        + str((datetime.now() + relativedelta(years=1)).year),
+        "is_current": "true",
+        "start_date": datetime.now().strftime("%Y-%m-%d"),
+        "end_date": "",
+        "group_id": school_data["id"],
+        "group_type": "school",
+        "user_id": data["user"]["id"],
+        "grade_id": data["grade_id"],
+    }
+    await enrollment_record.create_enrollment_record(
+        build_request(method="POST", body=enrollment_record_data)
+    )
 
-    > $BASE_URL/student/?student_id=1234
-    returns [{student_data}]
 
-    > $BASE_URL/student/?stream=PCB
-    returns [data_of_all_students_with_stream_PCB]
+async def create_auth_group_user_record(data, auth_group_name):
+    auth_group_data = auth_group.get_auth_group(
+        build_request(query_params={"name": auth_group_name})
+    )
+    group_data = group.get_group(
+        build_request(
+            query_params={"child_id": auth_group_data["id"], "type": "auth_group"}
+        )
+    )
 
-    > $BASE_URL/student/?student_id=student_id_with_stream_PCM&stream=PCB
+    await group_user.create_group_user(
+        build_request(
+            method="POST",
+            body={"group_id": group_data[0]["id"], "user_id": data["user"]["id"]},
+        )
+    )
 
-    returns {
-        "status_code": 404,
-        "detail": "No student found!",
-        "headers": null
+    enrollment_record_data = {
+        "academic_year": str(datetime.now().year)
+        + "-"
+        + str((datetime.now() + relativedelta(years=1)).year),
+        "is_current": "true",
+        "start_date": datetime.now().strftime("%Y-%m-%d"),
+        "end_date": "",
+        "group_id": auth_group_data["id"],
```

Comment:
same question.. group_data[0]["id"] ?

Reply:
No because auth_group API assumes that one only unique auth_group exists in the table. Which is true.